### PR TITLE
Introducing CEPTR for cantera to Pele C++ mechanisms

### PR DESCRIPTION
**Pull request by marchdf**
I got tired of supporting Pythia and Fuego not being able to parse mechanisms.

Here is a new mechanism converter. Notable things:
- converts Cantera yaml files to mechanism.{cpp,H}
- it is easy to get a cantera yaml file: `ck2yaml --input=mechanism.inp --thermo=therm.dat --transport=tran.dat --permissive` and I will be converting all our files to yaml in the near future
- you just `poetry run convert ${PATH_TO_CANTERA_YAML_FILE}`. I added a Lidryer and air example.
- no more pythia. I want a commit where both exist, so I will be removing that in a separate PR
- By using Cantera, we are using updated values of atomic weights. Pythia uses hard-coded numbers from 1969. So the variations we see in atomic weight will propagate across to other things (e.g. COFETA, etc). This will cause diffs across the codes. To revert back to the old atomic weights, one can add this to the yaml file:
```
elements:
- symbol: C
  atomic-weight: 12.01115
- symbol: O
  atomic-weight: 15.9994
- symbol: H
  atomic-weight: 1.00797
- symbol: N
  atomic-weight: 14.0067
- symbol: AR
  atomic-weight: 39.948
```
- For some mechanisms, cantera has fixed an ambiguity in the way three-body reactions are defined. This will lead to some reordering differences for some mechanisms (e.g. DRM19). More details at https://github.com/Cantera/cantera/issues/1251. This also causes some subtleties when computing the coefficients.
- In doing this, figured out that FORD was not dealt with properly in Fuego and was giving the wrong pre_exponential_factor value for the rates. This can be evident in looking at JL4 mechanism (reaction 0) and noticing that the current value is `39.1` when it should be `219875.458149426`. Verified with Cantera. This happens for mechanisms that have `reaction.orders` as a non-empty dict.
- Also,  reactions with `REV` in Chemkin files is transformed by `ck2yaml` into 2 separate reactions that are irreversible, e.g. NUIGalway:
```
H2O2+H<=>H2O+OH 2.410E+13 0.00 3.970E+03
REV/ 1.269E+08 1.31 7.141E+04 /
```
becomes
```
- equation: H2O2 + H => H2O + OH  # Reaction 31
  rate-constant: {A: 2.41e+13, b: 0.0, Ea: 3970.0}
- equation: H2O + OH => H2O2 + H  # Reaction 32
  rate-constant: {A: 1.269e+08, b: 1.31, Ea: 7.141e+04}
```

Things left to do
- [x] document how to setup and run
- [x] port over QSS (mostly done but I need help checking)
- [x] regenerate (and check!) all mechanisms to make sure that everything is ok
- [x] ensure ckinu and rmap are in there
- [x] convert all mechanism.inp chemkin stuff to yaml
- [x] add linting
- [x] add tests
- [x] check GPU build
- [x] regenerate all mechanisms

### CODE REVIEW

**Comment by marchdf** [comment]
Some fun stats. This PR should enable us to replace Pythia:
```
┏━━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━┳━━━━━━━┳━━━━━━┳━━━━━━━━━┳━━━━━━┓
┃ Language      ┃ Files ┃     % ┃  Code ┃    % ┃ Comment ┃    % ┃
┡━━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━╇━━━━━━━╇━━━━━━╇━━━━━━━━━╇━━━━━━┩
│ Python        │   496 │  60.0 │ 31290 │ 52.0 │   13920 │ 23.1 │
│ C++           │    35 │   4.2 │ 13283 │ 65.8 │    3966 │ 19.6 │
│ Objective-C++ │    91 │  11.0 │  3164 │ 74.4 │      29 │  0.7 │
│ FortranFixed  │     7 │   0.8 │  2432 │ 40.1 │    3612 │ 59.6 │
│ C             │    36 │   4.4 │   851 │ 40.7 │     713 │ 34.1 │
│ TOML          │     2 │   0.2 │   551 │ 86.2 │       0 │  0.0 │
│ HTML          │     2 │   0.2 │   128 │ 56.6 │      26 │ 11.5 │
│ Makefile      │     3 │   0.4 │    44 │ 50.0 │      19 │ 21.6 │
│ markdown      │     1 │   0.1 │    20 │ 55.6 │       0 │  0.0 │
│ Singularity   │     4 │   0.5 │     6 │  6.7 │      80 │ 89.9 │
│ __unknown__   │   127 │  15.4 │     0 │  0.0 │       0 │  0.0 │
│ __duplicate__ │    13 │   1.6 │     0 │  0.0 │       0 │  0.0 │
│ __binary__    │     9 │   1.1 │     0 │  0.0 │       0 │  0.0 │
├───────────────┼───────┼───────┼───────┼──────┼─────────┼──────┤
│ Sum           │   826 │ 100.0 │ 51769 │ 55.1 │   22365 │ 23.8 │
└───────────────┴───────┴───────┴───────┴──────┴─────────┴──────┘
```

with CEPTR:
```
┏━━━━━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━┳━━━━━━┳━━━━━━━┳━━━━━━━━━┳━━━━━┓
┃ Language         ┃ Files ┃     % ┃ Code ┃     % ┃ Comment ┃   % ┃
┡━━━━━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━╇━━━━━━╇━━━━━━━╇━━━━━━━━━╇━━━━━┩
│ Python           │    16 │  72.7 │ 7062 │  70.4 │     548 │ 5.5 │
│ TOML             │     2 │   9.1 │  932 │  84.7 │       0 │ 0.0 │
│ reStructuredText │     1 │   4.5 │    3 │ 100.0 │       0 │ 0.0 │
│ __unknown__      │     2 │   9.1 │    0 │   0.0 │       0 │ 0.0 │
│ __empty__        │     1 │   4.5 │    0 │   0.0 │       0 │ 0.0 │
├──────────────────┼───────┼───────┼──────┼───────┼─────────┼─────┤
│ Sum              │    22 │ 100.0 │ 7997 │  71.9 │     548 │ 4.9 │
└──────────────────┴───────┴───────┴──────┴───────┴─────────┴─────┘
```

**Comment by drummerdoc** [comment]
Excellent!  Happy to jettison Pythia when we're ready...it served its purpose well for years, but it's time to part ways.

**Comment by marchdf** [comment]
@malihass helped me check this with ignition delay of LiDryer:
Figure_1


We are checking other mechanisms as well.

**Comment by marchdf** [comment]
dodecane_lu:
Screen Shot 2022-05-17 at 8 46 25 PM

**Comment by marchdf** [comment]
dodecan_lu_qss:
Screen Shot 2022-05-17 at 8 48 02 PM

**Review by jrood-nrel — APPROVED** [review]

**Comment by marchdf** [comment]
Any objections to merging this?

File: Support/Fuego/Mechanism/Models/Aromatic_KrNara/mechanism.yaml [hunk 1]
```diff
@@ -0,0 +1,5090 @@
...
+    A1C2H3, A1C2H, A1C2H*, A1C2H3*, A2-, A2, A2*, A2C2H2A, A2C2H2B, A2C2HA,
+    A2C2HB, A2C2HA*, A2C2HB*, A2R5, A2R5-, A2R5C2H2, A2R5C2H, A2R5C2H*,
+    P2, P2-, A3-, A3, A3*, A3R5-, A3R5, A4, A4-, A4R5, FLTN, C5H6, C5H5,
+    T-C5H5O, C5H4O, S-C5H5O, C9H8, C9H7, A1CH2, C9H6O, O-C6H4, A1CH3, A1OH,
+    HOA1CH3, OA1CH3, A1CH2O, A1CH2OH, A1CHO, A1O, A1CH3*, A1C2H4, A1C2H5,
+    C8H9O2, C8H8OOH, OC8H7OOH, A1CH3CH3, A1CH3CH2, A1CH3CHO, A2CH3, A1CHOCH2,
+    A1CHOCHO, A2OH, A2CH2, A2CH2O, A2CHO, A2O, OC6H4O, N2]
+  kinetics: gas
+  transport: mixture-averaged
+  state: {T: 300.0, P: 1 atm}
```
Comment (malihass) [inline Support/Fuego/Mechanism/Models/Aromatic_KrNara/mechanism.yaml:33]: Isn't it a bit odd to have the state of the mixture in the mechanism file ?

File: Testing/Exec/CeptrTest/CanteraSimulation/homogenous-reactor.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+"""Run a homegenous reactor in Cantera."""
```
Comment (malihass) [inline Testing/Exec/CeptrTest/CanteraSimulation/homogenous-reactor.py:1]: typo

File: Testing/Exec/CeptrTest/PelePhysicsSimulation/utils/plotFunctions.H [hunk 3]
```diff
@@ -0,0 +1,29 @@
+void
```
Comment (malihass) [inline Testing/Exec/CeptrTest/PelePhysicsSimulation/utils/plotFunctions.H:1]: I think you can clean up the `utils` folder. 
I think you only need the `initFunctions.H` and the `reactFunctions.H` Also those are in the `IgnitionDelay` test case, so maybe we can remove some redundancy here?

**Review by malihass — CHANGES_REQUESTED** [review]

File: Support/Fuego/Mechanism/Models/Aromatic_KrNara/mechanism.yaml (hunk 1 above)
Comment (baperry2) [reply Support/Fuego/Mechanism/Models/Aromatic_KrNara/mechanism.yaml:33]: It is a bit odd but these are put in automatically by ck2yaml. I believe they just give the default conditions when creating a Solution in cantera with the mechanism. The mixture state and transport can be safely removed, but I don't think there is a need to do so.

**Pull request merged**